### PR TITLE
fix: ability to create 404 page via Registry UI

### PR DESCRIPTION
**Pull request by nightnei**

### CODE REVIEW

File: registry/tests/appRoutes.spec.ts [hunk 1]
```diff
@@ -239,27 +239,27 @@ describe(`Tests ${example.url}`, () => {
...
-                }
-
-                expect(response2.status).equal(500);
-                expect(response2.text).to.include('UNIQUE constraint failed: routes.orderPos');
-            } finally {
-                routeId1 && await request.delete(example.url + routeId1);
-                routeId2 && await request.delete(example.url + routeId2);
-            }
-        });
+        // it('should not create record with the same orderPos', async () => {
```
Comment (StyleT) [inline registry/tests/appRoutes.spec.ts:242]: Let's remove or uncomment

File: registry/server/migrations/20210426163602_routes_orderPosIsNullable.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
+import * as Knex from "knex";
+
+export async function up(knex: Knex): Promise<any> {
+    if (isMySQL(knex)) {
+        return knex.schema.alterTable('routes', table => {
+            table.integer('orderPos', 10).nullable().alter(); // alter table don't affect only uniq constraint, so you don't need and it's disallowed to set unique('routes_orderpos_unique') here
+            table.string('route', 255).nullable().alter();
+            table.boolean('next').nullable().alter();
```
Comment (StyleT) [inline registry/server/migrations/20210426163602_routes_orderPosIsNullable.ts:8]: Why would you make it null? Just fallback to false

File: registry/server/migrations/20210426163602_routes_orderPosIsNullable.ts [hunk 3]
```diff
@@ -0,0 +1,84 @@
+import * as Knex from "knex";
+
+export async function up(knex: Knex): Promise<any> {
+    if (isMySQL(knex)) {
+        return knex.schema.alterTable('routes', table => {
+            table.integer('orderPos', 10).nullable().alter(); // alter table don't affect only uniq constraint, so you don't need and it's disallowed to set unique('routes_orderpos_unique') here
+            table.string('route', 255).nullable().alter();
```
Comment (StyleT) [inline registry/server/migrations/20210426163602_routes_orderPosIsNullable.ts:7]: discussed opts:
- rm specialRole & put special ID to route
- keep for now as is

Comment (StyleT) [reply registry/server/migrations/20210426163602_routes_orderPosIsNullable.ts:7]: special:404

File: registry/server/appRoutes/interfaces/index.ts [hunk 4]
```diff
@@ -45,7 +45,7 @@ const commonAppRoute = {
     specialRole: Joi.string().valid('404'),
     orderPos: Joi.number(),
     route: Joi.string().trim().max(255),
-    next: Joi.bool().default(false),
+    next: Joi.bool(),
```
Comment (StyleT) [inline registry/server/appRoutes/interfaces/index.ts:48]: this change is a breaking one

File: registry/tests/appRoutes.spec.ts (hunk 1 above)
Comment (nightnei) [reply registry/tests/appRoutes.spec.ts]: it was commented temporary

File: registry/server/migrations/20210426163602_routes_orderPosIsNullable.ts (hunk 2 above)
Comment (nightnei) [reply registry/server/migrations/20210426163602_routes_orderPosIsNullable.ts]: reverted

File: registry/server/migrations/20210426163602_routes_orderPosIsNullable.ts (hunk 3 above)
Comment (nightnei) [reply registry/server/migrations/20210426163602_routes_orderPosIsNullable.ts]: done

File: registry/server/appRoutes/interfaces/index.ts (hunk 4 above)
Comment (nightnei) [reply registry/server/appRoutes/interfaces/index.ts]: reverted

File: ilc/server/tailor/fetch-template.js [hunk 5]
```diff
@@ -24,7 +24,7 @@ module.exports = (configsInjector, newrelic, registryService) => async (
         throw new Error('Can\'t match route base template to config map');
     }
 
-    const routeName = currRoute.route.replace(/^\/(.+)/, '$1');
+    const routeName = currRoute.route?.replace(/^\/(.+)/, '$1');
```
Comment (StyleT) [inline ilc/server/tailor/fetch-template.js:27]: Let's make sure that we always name NR transaction

**Review by StyleT — APPROVED** [review]

**Pull request merged**